From 76fc3630455198b5c54eb28684eaf51f2da3ebe7 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Wed, 26 Jun 2019 00:37:08 +0800
Subject: [PATCH 2/4] platform/x86: sony-laptop: add new VAIO ACPI ID

As VAIO is sold by Sony, new VAIO laptops do not use the original
"SNY5001" ACPI ID for notebook control device. Instead a "VAI0001" ACPI
ID is used. The interface of "VAI0001" is still the same with "SNY5001".

So add the "VAI0001" ID to the sony-laptop driver, to drive the new
VAIO laptops' notebook control device.

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
---
 drivers/platform/x86/sony-laptop.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/platform/x86/sony-laptop.c b/drivers/platform/x86/sony-laptop.c
index fb088dd8529e..459fb23a3e3d 100644
--- a/drivers/platform/x86/sony-laptop.c
+++ b/drivers/platform/x86/sony-laptop.c
@@ -65,6 +65,7 @@ do {						\
 
 #define SONY_NC_CLASS		"sony-nc"
 #define SONY_NC_HID		"SNY5001"
+#define VAIO_NC_HID		"VAI0001"
 #define SONY_NC_DRIVER_NAME	"Sony Notebook Control Driver"
 
 #define SONY_PIC_CLASS		"sony-pic"
@@ -3301,6 +3302,7 @@ static int sony_nc_remove(struct acpi_device *device)
 
 static const struct acpi_device_id sony_device_ids[] = {
 	{SONY_NC_HID, 0},
+	{VAIO_NC_HID, 0},
 	{SONY_PIC_HID, 0},
 	{"", 0},
 };
@@ -3308,6 +3310,7 @@ MODULE_DEVICE_TABLE(acpi, sony_device_ids);
 
 static const struct acpi_device_id sony_nc_device_ids[] = {
 	{SONY_NC_HID, 0},
+	{VAIO_NC_HID, 0},
 	{"", 0},
 };
 
-- 
2.21.0

